drivers/lpsxxx/lpsxxx.c

```diff
@@ -0,0 +1,256 @@
+
+/**
+ * @brief temperature base value and divider for norming temperature output
+ */
+#if MODULE_LPS331AP || MODULE_LPS25HB
+#define TEMP_BASE           (42.5f)
+#define TEMP_DIVIDER        (480U)
+#else
+#define TEMP_BASE           (0.0f)
+#define TEMP_DIVIDER        (100U)
+#endif
+
+#define DEV_I2C      (dev->params.i2c)
+#define DEV_ADDR     (dev->params.addr)
+#define DEV_RATE     (dev->params.rate)
+
+int lpsxxx_init(lpsxxx_t *dev, const lpsxxx_params_t * params)
+{
+    dev->params = *params;
+
+    /* Acquire exclusive access to the bus. */
+    i2c_acquire(DEV_I2C);
+    uint8_t id;
+    if (i2c_read_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_WHO_AM_I, &id, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: cannot read WHO_AM_I register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+
+    if (id != LPSXXX_WHO_AM_I) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: not a valid device (got %02X, expected %02X)\n",
+              id, LPSXXX_WHO_AM_I);
+        return -LPSXXX_ERR_NODEV;
+    }
+
+    uint8_t tmp;
+
+#if MODULE_LPS22HB
+    if (i2c_read_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_CTRL_REG2, &tmp, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: cannot read LPSXXX_REG_CTRL_REG2 register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+
+    /* Disable automatic increment of register address during byte access
+       (recommanded in datasheet (section 9.6 CTRL_REG2) */
+    tmp &= ~LPSXXX_CTRL_REG2_ID_ADD_INC;
+
+    DEBUG("[lpsxxx] init: update reg2, %02X\n", tmp);
+
+    if (i2c_write_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_CTRL_REG2, tmp, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: cannot write in CTRL_REG2 register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+#endif
+
+    /* configure device, for simple operation only CTRL_REG1 needs to be touched */
+#if MODULE_LPS331AP
+    tmp = LPSXXX_CTRL_REG1_DBDU | LPSXXX_CTRL_REG1_PD |
+            (DEV_RATE << LPSXXX_CTRL_REG1_ODR_POS);
+#elif MODULE_LPS25HB
+    tmp = LPSXXX_CTRL_REG1_BDU | LPSXXX_CTRL_REG1_PD |
+            (DEV_RATE << LPSXXX_CTRL_REG1_ODR_POS);
+#elif MODULE_LPS22HB
+    tmp = LPSXXX_CTRL_REG1_EN_LPFP | /* Low-pass filter configuration: ODR/9 */
+            LPSXXX_CTRL_REG1_BDU | (DEV_RATE << LPSXXX_CTRL_REG1_ODR_POS);
+#endif
+
+    DEBUG("[lpsxxx] init: update reg1, value: %02X\n", tmp);
+
+    if (i2c_write_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_CTRL_REG1, tmp, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[lpsxxx] init: cannot write in CTRL_REG1 register\n");
+        return -LPSXXX_ERR_I2C;
+    }
+
+    i2c_release(DEV_I2C);
+
+    DEBUG("[lpsxxx] initialization successful\n");
+    return LPSXXX_OK;
+}
+
+int lpsxxx_read_temp(const lpsxxx_t *dev, int16_t *temp)
+{
+    uint8_t tmp;
+    int16_t val = 0;
+    float res = TEMP_BASE;      /* reference value -> see datasheet */
+
+    i2c_acquire(DEV_I2C);
+    if (i2c_read_reg(DEV_I2C, DEV_ADDR, LPSXXX_REG_TEMP_OUT_L, &tmp, 0) < 0) {
```

Comment:
Hmmm I see the point...

I am now curious if there is either a problem with an update that could occur after reading the lsb or if updates just don't work if only reading temperature since 
`1. To guarantee the correct behavior of BDU feature, PRESS_OUT_H (2Ah) must be the last address read.`

Reply:
@MrKevinWeiss, I have just tested it and there's no problem on both lps25hb and lps22hb.

Reply:
even if I don't read the pressure measure at all.

Reply:
Hmm OK, then do we worry about the MSB updating after the LSB read?

Reply:
Or can we assume the BDU takes care of that?  If so I suppose I am happy and can start testing with iotlabs

Reply:
> can we assume the BDU takes care of that?

I would say so.